### PR TITLE
Draft yaml configuration file for dart-lang for autosubmit support

### PR DESCRIPTION
**Pull request by ricardoamador**
Adding a global config file in order to support dart-lang with the autosubmit bot.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: autosubmit/dart-lang/autosubmit.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+default_branch: master
```
Comment (devoncarew) [inline autosubmit/dart-lang/autosubmit.yml:5]: Our repos are on a mix of master and main (a little over 20% use `main` as the default branch name).

Comment (devoncarew) [reply autosubmit/dart-lang/autosubmit.yml:5]: But I suppose we're just worried about `dart-lang/pubspec_parse` for now?

File: autosubmit/dart-lang/autosubmit.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+issues_repository:
+  owner: dart-lang
+  repo: pubspec_parse
+auto_approval_accounts:
+  - dependabot[bot]
+  - dependabot
+  - DartDevtoolWorkflowBot
```
Comment (devoncarew) [inline autosubmit/dart-lang/autosubmit.yml:12]: We probably don't need this bot approved?

File: autosubmit/dart-lang/autosubmit.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+approving_reviews: 2
+# need to provide this if possible or we need to accept that there is no group
+approval_group: flutter-hackers
```
Comment (devoncarew) [inline autosubmit/dart-lang/autosubmit.yml:15]: `dart-team`?

File: autosubmit/dart-lang/autosubmit.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+run_ci: true
+support_no_review_revert: true
+required_checkruns:
+  - "cla/google"
+  - "analyze (dev)"
```
Comment (devoncarew) [inline autosubmit/dart-lang/autosubmit.yml:20]: We have various names for the workflows for different repos; and some repos kick off as many as several dozen workflow runs per PR. We'll likely need a way to indicate that all checks are required to pass.

**Review by devoncarew — DISMISSED** [review]
lgtm for setting up this one repo; we will likely need some iteration in order to support the various dart-lang repos.

File: autosubmit/dart-lang/autosubmit.yml (hunk 3 above)
Comment (ricardoamador) [reply autosubmit/dart-lang/autosubmit.yml:15]: Yeah this is what I was looking for. Okay I will make the change.

File: autosubmit/dart-lang/autosubmit.yml (hunk 4 above)
Comment (ricardoamador) [reply autosubmit/dart-lang/autosubmit.yml:20]: Ah good to know. We will definitely needs this.

File: autosubmit/dart-lang/autosubmit.yml (hunk 1 above)
Comment (ricardoamador) [reply autosubmit/dart-lang/autosubmit.yml:5]: yeah just for now we are including this as it make it easier for dev but will be removed at a later time.

File: autosubmit/dart-lang/autosubmit.yml (hunk 2 above)
Comment (ricardoamador) [reply autosubmit/dart-lang/autosubmit.yml:12]: Okay, will remove.

**Review by devoncarew — APPROVED** [review]

**Pull request merged**